File: java/src/main/java/com/pos/integration/model/refund/RefundRequest.java
package com.pos.integration.model.refund;

import com.pos.integration.model.common.Currency;
import com.pos.integration.model.common.PosCredential;
import lombok.Builder;
import lombok.Getter;

import java.math.BigDecimal;


@Getter
@Builder
public class RefundRequest {

    private final PosCredential posCredential;

    private final BigDecimal amount;
    private final Currency currency;
    private final String refundReferenceNumber;
}

File: kotlin/src/main/kotlin/com/trendyol/posintegrationapi/model/refund/RefundRequest.kt
package com.trendyol.posintegrationapi.model.refund

import com.trendyol.posintegrationapi.model.common.Currency
import com.trendyol.posintegrationapi.model.common.PosCredential
import java.math.BigDecimal

open class RefundRequest {
    var posCredential: PosCredential? = null
    var amount: BigDecimal? = null
    var currency: Currency? = null
    var refundReferenceNumber: String? = null
}
